src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -251,7 +252,8 @@ using MoreTypes = boost::variant<
         ISRLU<arma::mat, arma::mat>*,
         BicubicInterpolation<arma::mat, arma::mat>*,
         NearestInterpolation<arma::mat, arma::mat>*,
-        GroupNorm<arma::mat, arma::mat>*
+        GroupNorm<arma::mat, arma::mat>*,
+        SQNL<arma::mat, arma::mat>*
```

Comment:
You would need to define the class too:
```c++
template <typename InputDataType, typename OutputDataType> class SQNL;
```

Reply:
Just a thing I noticed, I have not seen any other activation layer classes being defined anywhere in this file

Reply:
I could be wrong in that case, @zoq can you help ?

Reply:
Seems like there should be a class definition for every module that could be used to construct a model, shouldn't there be? If yes, then where are they?

Reply:
Most of the activation functions use the base layer, here is one example:

https://github.com/mlpack/mlpack/blob/e02ab3be544694294d2f73bd12a98d0d162ef3af/src/mlpack/methods/ann/layer/base_layer.hpp#L151-L152

which is why you don't see a lot of activation functions there, in some cases the activation function is a separate file as well, one example:

https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/layer_types.hpp#L290